copycat/api/src/main/java/org/apache/kafka/copycat/data/ISchema.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.copycat.data;
+
+import java.util.List;
+
+/**
+ * @see Schema
+ * @see SchemaBuilder
+ */
+public interface ISchema {
```

Comment:
The convention elsewhere would be the interface gets the good name (e.g. Schema) and the impl gets a system prefix CopycatSchema, we haven't done I- prefixes elsewhere. But actually are there really going to be multiple implementations of Schema? I.e. for Avro would we write an adaptor or would we just take the fields out of the AvroRecord and put them into the CopycatSchema instance?